lab-judy/test/person-route-test.js

```diff
@@ -0,0 +1,234 @@
+      Person.createPerson(examplePerson)
+      .then(person => {
+        this.tempPerson = person;
+        done();
+      })
+      .catch( err => done(err));
+    });
+    //clean up the mocked person
+    after(done => {
+      Person.deletePerson(this.tempPerson.id)
+      .then(() => done())
+      .catch(err => done(err));
+    });
+
+    it('should return 200 for request with valid body', done => {
+      let updatedPerson = {name: 'updated', age: '45'};
+      request.put(`${url}/api/person/${this.tempPerson.id}`)
+      .send(updatedPerson)
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.id).to.equal(this.tempPerson.id);
+        for (var key in updatedPerson){
+          expect(res.body[key]).to.equal(updatedPerson[key]);
+        }
+        done();
+      });
+    });
+  });
+  // * TODO: `POST` - test 200, response body like  `{<data>}` for a post request with a valid body
+  describe('testing for POST request with valid body', function(){
+    before(done => {
+      Person.createPerson(examplePerson)
+      .then(person => {
+        this.tempPerson = person;
+        done();
+      })
+      .catch( err => done(err));
+    });
+    //clean up the mocked person
+    after(done => {
+      Person.deletePerson(this.tempPerson.id)
+      .then(() => done())
+      .catch(err => done(err));
+    });
+    it('should return a 200 status for post request with valid body', done => {
+      request.post(`${url}/api/person`)
+      .send(examplePerson)
+      .end((err, res) => {
+        if (err) return done (err);
+        expect(res.status).to.equal(200);
+        expect(!!res.body.id).to.equal(true);
+        expect(res.body.name).to.equal(examplePerson.name);
+        expect(res.body.age).to.equal(examplePerson.age);
+        this.tempPerson = res.body;
+        done();
+      });
+
+    });
+  });
+  // `POST` - test 400, responds with 'bad request' for if no `body provided` or `invalid body`
+  describe('testing post for bad request if no body provided', function(){
+    before(done => {
+      Person.createPerson(examplePerson)
+      .then(person => {
+        this.tempPerson = person;
+        done();
+      })
+      .catch(err => done(err));
+    });
+    after(done => {
+      Person.deletePerson(this.tempPerson.id)
+      .then( () => done())
+      .catch(err => done(err));
+    });
+    it('should return 400 with bad post request for invalid body provided', done =>{
+      request.post(`${url}/api/person/${this.tempPerson.id}`)
+      .set('Content-Type','application/json')
+      .send('invalid body')
+      .end((err, res) => {
+        expect(res.status).to.equal(404);
+        done();
+      });
+    });
+  });
+//DELETE - test 404, for a DELETE request with an invalid or missing id
+//404 for missing id because DELETE /api/<simple-resource-name>/ is not a route
+  describe('testing DELETE for request with invalid/missing id', function(){
+    before(done => {
```

Comment:
No need to create a person if you're testing a 404 -- nothing will be deleted if the request is bad or the person is not found.